### PR TITLE
feat: autogenerated sidebar structure

### LINKED ISSUE #150: chore: sidebars / footer links configured

### PR DESCRIPTION
**Pull request by mfw78**
# Description

This PR supercedes #180

# What is **NOT** covered by this PR

1. Movement of documentation between sections.
2. Image changes (CoW Protocol vs CoW DAO)
3. Some sub-sections may still have lower-case (these are generally OLD files that are to be migrated still).

# Changes

- [x] Applies re-organization of sidebar menu
- [x] Adds explicit category definitions and order
- [x] Sets autogenerated sidebar based on file structure in docs folder
- [x] Maintains `cow-protocol` in separate directory to allow for documentation by product
- [x] Explicitly move some old files to 'draft' status so they don't show in production

## Related Issues

Fixes #150

### CODE REVIEW

File: docs/cow-protocol/reference/_category_.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "position": 3,
+    "label": "📚 Technical reference",
+    "collapsible": true,
+    "collapsed": true,
+    "link": {
+        "type": "generated-index",
+        "title": "📚 Technical reference"
+    },
+    "customProps": {
+        "description": "This is the technical reference section."
```
Comment (anxolin) [inline docs/cow-protocol/reference/_category_.json:11]: I check the page. I didn't see where this description is being shown. Should it be displayed in the content?

File: docs/cow-protocol/tutorials/cow-swap/cancel.mdx [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ---
-sidebar_position: 3
+sidebar_position: 5
```
Comment (anxolin) [inline docs/cow-protocol/tutorials/cow-swap/cancel.mdx:2]: nit: i thought after Swap made sense as they are the basic flows, but not strong preference.
Limit orders, twap and native are enhancements

**Review by marshymarsh — APPROVED** [review]
Looks great!

**Review by anxolin — APPROVED** [review]

File: docs/cow-protocol/reference/_category_.json (hunk 1 above)
Comment (mfw78) [reply docs/cow-protocol/reference/_category_.json:11]: No idea, this was a thing in the previous PR from @fairlighteth :sweat_smile: I can see it in https://docusaurus.io/docs/sidebar/autogenerated#category-item-metadata

**Pull request merged**